sstables/sstables.cc

```diff
@@ -1535,6 +1535,35 @@ future<foreign_sstable_open_info> sstable::get_open_info() & {
     });
 }
 
+future<std::vector<shard_id>>
+sstable::compute_owner_shards(const io_priority_class& pc) {
+    co_await read_scylla_metadata(pc);
```

Comment:
the compute function should just work on a reference to the data structure in memory.
The caller should load it, not the compute function

Reply:
i.e. we should refactor the logic out of compute_shards_for_this_sstable()

Reply:
You have a good point.

Today, if you want to get the shards owning a SSTable, you do;
```
sstable.load();
sstable.get_shards_for_this_sstable();
```

sstable::load() loads the sharding metadata, interpret it and assign result to sstable::_shards.

So it makes sense to have:
`future<> sstable::load_owner_shards();`

which loads the minimum from disk, in order to set sstable::_shards (as sstable::load() does).

And then we can still piggyback on sstable::get_shards_for_this_sstable() to retrieve the list of shards